br: add docs for PiTR (#9823)
pingcap/docs: br/point-in-time-recovery.md

---
title: Point-in-Time Recovery
summary: Learn the design, capabilities, and architecture of Point-in-Time Recovery (PITR).
---

# Point-in-Time Recovery

Point-in-Time Recovery (PITR) allows you to restore a snapshot of a TiDB cluster to a new cluster from any given time point in the past. In v6.2.0, TiDB introduces PITR in [Backup & Restore](/br/backup-and-restore-overview.md) (BR).

You can use PITR to meet the following business requirements:

> - Reduce the RPO of disaster recovery to less than 20 minutes.

Reviewer: ```suggestion
- Reduce the Recovery Point Objective (RPO) of disaster recovery to less than 20 minutes.
```

> - Handle the cases of incorrect writes from applications by rolling back data to before the error event.

Reviewer: ```suggestion
- Handle the cases of incorrect writes from applications by rolling back data to a time point before the error event.
```

- Perform history data auditing to meet the requirements of laws and regulations.

This document introduces the design, capabilities, and architecture of PITR. If you need to learn how to use PITR, refer to [PITR Usage Scenarios](/br/pitr-usage.md).

## Use PITR in your business

[BR](/br/backup-and-restore-overview.md) is the tool for using the PITR feature. With BR, you can perform all operations of PITR, including data backup (snapshot backup and log backup), one-click restoration to a specified time point, and backup data management.

The following are the procedures of using PITR in your business:

> ![Point-in-Time Recovery](/media/br/pitr-usage.png)

Reviewer: There are some errors in the diagram:
- 'these data' should be 'the following data'
- 'logs backups' should be 'log backups'
- 'PiTR' should 'PITR'

Author: fixed in https://github.com/pingcap/docs-cn/pull/10841

### Back up data

To achieve PITR, you need to perform the following backup tasks:

- Start a log backup task. You can run the `br log start` command to start a log backup task. This task runs in the background of your TiDB cluster and automatically backs up the change log of KV storage to the backup storage.

> - Perform [snapshot (full) backup](/br/br-usage-backup.md#back-up-tidb-cluster-snapshots) regularly. You can run the `br backup full` command to back up the cluster snapshot to the backup storage at, for example, 00:00 every day.

Reviewer: ```suggestion
- Perform [snapshot (full) backup](/br/br-usage-backup.md#back-up-tidb-cluster-snapshots) regularly. You can run the `br backup full` command to back up the cluster snapshot to the backup storage at a specified time, for example, 00:00 every day.
```

- On each TiKV node, PITR can restore snapshot data at 280 GB/h and log data at 30 GB/h.
- With PITR, the RPO of disaster recovery is less than 20 minutes. Depending on the data size to be restored, the RTO varies from several minutes to several hours.
- BR deletes outdated log backup data at a speed of 600 GB/h.

> **Note:**
>
> - The preceding functional specification is based on test results from the following two testing scenarios. The actual data may be different.
> - Snapshot data restoration speed = Snapshot data size / (duration * the number of TiKV nodes)
> - Log data restoration speed = Restored log data size / (duration * the number of TiKV nodes)

Testing scenario 1 (on [TiDB Cloud](https://tidbcloud.com)):

- The number of TiKV nodes (8 core, 16 GB memory): 21
- The number of Regions: 183,000
- New log created in the cluster: 10 GB/h
- Write (insert/update/delete) QPS: 10,000

Testing scenario 2 (on-premises):

- The number of TiKV nodes (8 core, 64 GB memory): 6
- The number of Regions: 50,000
- New log created in the cluster: 10 GB/h
- Write (insert/update/delete) QPS: 10,000

## Limitations

- A single cluster can only start one log backup task.

> - You can only restore data to an empty cluster. To avoid impact on the services and data of the cluster, you cannot perform PITR in-place or on a non-empty cluster.

Reviewer: ```suggestion
- You can only restore data to an empty cluster. To avoid impact on the services and data of the cluster, do not perform PITR in-place or on a non-empty cluster.
```

- You can use Amazon S3 or a shared filesystem (such as NFS) to store the backup data. Currently, GCS and Azure Blob Storage are not supported.
- You can only perform PiTR on cluster level. Database-level and table-level PiTR are not supported.
- You cannot restore data in the user tables or the privilege tables.
- If the backup cluster has a TiFlash replica, after you perform PiTR, the restoration cluster does not contain the data in the TiFlash replica. To restore data from the TiFlash replica, you need to [manually configure the TiFlash replica in the schema or the table](/br/pitr-troubleshoot.md#在使用-br-restore-point-命令恢复下游集群后无法从-tiflash-引擎中查询到数据该如何处理). //TODO

> - If the upstream database uses TiDB Lightning's physical import mode to import data, the data cannot be backed up in log backup. It is recommended to perform a full backup after the data import. For details, refer to [上游数据库使用 TiDB Lightning Physical 方式导入数据的恢复](/br/pitr-known-issues.md#上游数据库使用-tidb-lightning-physical-方式导入数据导致无法使用日志备份功能). //TODO

Author: aligned with https://github.com/pingcap/docs-cn/pull/10769

> - During the backup process, you cannot exchange partition. For details, refer to [Executing the Exchange Partition DDL during PiTR recovery](/br/pitr-troubleshoot.md#what-should-i-do-if-an-error-occurs-when-executing-the-exchange-partition-ddl-during-pitr-recovery).

Author: aligned with https://github.com/pingcap/docs-cn/pull/10769

> - During the backup process, you cannot exchange partition. For details, refer to [Executing the Exchange Partition DDL during PITR recovery](/br/pitr-troubleshoot.md#what-should-i-do-if-an-error-occurs-when-executing-the-exchange-partition-ddl-during-pitr-recovery).

Reviewer: ```suggestion
- During the backup process, do not exchange partition. For details, refer to [Executing the Exchange Partition DDL during PITR recovery](/br/pitr-troubleshoot.md#what-should-i-do-if-an-error-occurs-when-executing-the-exchange-partition-ddl-during-pitr-recovery).
```

> - You cannot restore the log backup data of a certain time period repeatedly. If you restore the log backup data of a range `[t1=10, t2=20)` repeatedly, the restored data might be inconsistent.

Reviewer: ```suggestion
- Do not restore the log backup data of a certain time period repeatedly. If you restore the log backup data of a range `[t1=10, t2=20)` repeatedly, the restored data might be inconsistent.
```
